### PR TITLE
[Master] Added docs  password recovery via preferred channel

### PR DESCRIPTION
**Pull request by divyaamunugama**
Related issue: https://github.com/wso2/product-is/issues/15035

### CODE REVIEW

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 1]
```diff
@@ -0,0 +1,534 @@
+# Password Recovery Via User Preferred Notification Channel
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:1]:
```suggestion
# Password Recovery Via User-Preferred Notification Channel
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+
+This section guides you through setting up password recovery for users to recover a lost or forgotten password.
+
+!!! Note
+    This feature is only available via Account Recovery REST APIs. Currently, WSO2 IS does not support this feature via the User Portal.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:6]: user portal?

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+
+## Password recovery flow
+
+The password recovery for of WSO2 Identity Server is as follows:
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:10]:
```suggestion
The password recovery flow of WSO2 Identity Server is as follows:
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+3. The user selects a notification channel.
+4. The server sends the recovery notification to the user via the preferred notification channel.
+5. If the preferred channel is:
+    - **Email**: The user can click the reset link sent to the registered email address and reset the password.
+    - **SMS**: The user provides the received One-Time Password (OTP) and reset the password.
+
+The following sections walk you through configuring and trying out password recovery via SMS and Email channels.
+
+??? Warning "If you are migrating from IS 5.10.0 or above"
+    If you have migrated from IS 5.10.0 or above, ensure that the `IdentityMgtEventListener` with the `orderId=50` is set to `false` and that the Identity Listeners with `orderId=95` and `orderId=97` are set to `true` in the `<IS_HOME>/repository/conf/deployment.toml` file.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:23]:
```suggestion
    If you have migrated from IS 5.10.0 or above, ensure that the `IdentityMgtEventListener` with the `orderId=50` is set to `false` and that the identity listeners with `orderId=95` and `orderId=97` are set to `true` in the `<IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+    [event.default_listener.governance_identity_store]
+    priority= "97"
+    enable = true
+    ```
+
+## Set up password recovery
+
+Follow the steps given below to recover a user in the super tenant (i.e., `carbon.super`).
+
+1. Add the following properties to the `deployment.toml` file in the `IS_HOME/repository/conf` folder to configure the WSO2 Identity Server to send confirmation emails.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:43]:
```suggestion
1. Add the following properties to the `deployment.toml` file in the `IS_HOME/repository/conf` folder to configure WSO2 Identity Server to send confirmation emails.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+    enable_start_tls= true
+    enable_authentication= true
+    ```
+
+    !!! Note
+        - Add this configuration only if you wish to configure WSO2 Identity Server to send confirmation emails. See [Configure the Email Sending Module]({{base_path}}/deploy/configure-email-sending/) for more information
+        - Alternatively, you can use your own email managing mechanism.
+
+
+2. Add the following properties to the `deployment.toml`.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:61]:
```suggestion
2. Add the following properties to the `deployment.toml` file.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+    resend_code_validity=5
+    
+    [identity_mgt.password_reset_sms]
+    sms_otp_validity=2
+    ```
+
+    <table>
+       <tr>
+           <th>Property Name</th>
+           <th>Functionality</th>
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:75]:
```suggestion
           <th>Description</th>
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+       </tr>
+       <tr>
+          <td><code>recovery_code_validity</code></td>
+          <td>
+            <ul>
+                <li>Validity period of the recovery code given after initiating username/ password
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:81]: Should this be part of the list? Looks like this is a description followed by the unit and default value.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+                recovery.
+                <li>Unit : <code>minutes</code>
+                <li>Default value : <code>1 minute</code>
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:84]: Is it `1 minute` or just `1`.
This should be the exact value.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+            </ul>
+            <div class="admonition note">
+                <p class="admonition-title">note</p>
+                <p>
+                    If you have configured Username Recovery Via User Preferred Notification Channel, the above field will already be configured.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:89]:
```suggestion
                    If you have configured username recovery via the user-preferred notification channel, the above field is already configured.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+                </p>
+            </div>
+          </td>
+       </tr>
+       <tr>
+          <td><code>resend_code_validity</code></td>
+          <td>
+            <ul>
+                <li>Validity period of the recovery code given after initiating password recovery.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:98]: same comment as before

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+                <li>Unit : <code>minutes</code>
+                <li>Default value : <code>1 minute</code>
+            </ul>
+          </td>
+       </tr>
+       <tr>
+         <td><code>sms_otp_validity</code></td>
+         <td>
+           <ul>
+                <li>Validity period of SMS OTP when the selected channel is <code>SMS</code>.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:108]: same comment as before.. 
Please correct me if I am wrong. Don't see how this is a list.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 13]
```diff
@@ -0,0 +1,534 @@
...
+                <li>Unit : <code>minutes</code>
+                <li>Default value : <code>1 minute</code>
+           </ul>
+         </td>
+      </tr>
+    </table>
+
+3. Add an event publisher to `<IS_HOME/repository/deployment/server/eventpublishers`. You can use the following sample publisher to call a REST Service to send confirmation codes. This sample uses the `http output adapter`
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:116]:
```suggestion
3. Add an event publisher to `<IS_HOME/repository/deployment/server/eventpublishers`. You can use the following sample publisher to call a REST service to send confirmation codes. This sample uses the `http output adapter`
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 14]
```diff
@@ -0,0 +1,534 @@
...
+                    }</inline>
+            </mapping>
+            <to eventAdapterType="http">
+                <property name="http.client.method">httpPost</property>
+                <property name="http.url">https://rest.nexmo.com/sms/json</property>
+            </to>
+        </eventPublisher>
+        ```
+
+        This publisher uses NEXMO as the SMS REST service provider. For more information on writing a custom http event publisher, see [HTTP Event Publisher](https://docs.wso2.com/display/DAS300/HTTP+Event+Publisher).
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:139]:
```suggestion
        This publisher uses NEXMO as the SMS REST service provider. For more information on writing a custom HTTP event publisher, see [HTTP Event Publisher](https://docs.wso2.com/display/DAS300/HTTP+Event+Publisher).  
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 15]
```diff
@@ -0,0 +1,534 @@
...
+                <property name="http.client.method">httpPost</property>
+                <property name="http.url">https://rest.nexmo.com/sms/json</property>
+            </to>
+        </eventPublisher>
+        ```
+
+        This publisher uses NEXMO as the SMS REST service provider. For more information on writing a custom http event publisher, see [HTTP Event Publisher](https://docs.wso2.com/display/DAS300/HTTP+Event+Publisher).  
+
+    !!! note
+        If an Identity Server instance is already running, restart it to apply the above configurations.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:142]:
```suggestion
        If a WSO2 IS instance is already running, restart it to apply the above configurations.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 16]
```diff
@@ -0,0 +1,534 @@
...
+
+4. On the management console, go to **Identity Providers** > **Resident** > **Account Management**, expand the **Account Recovery**, and configure the following properties.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:144]:
```suggestion
4. On the management console, go to **Identity Providers** > **Resident** > **Account Management**, expand **Account Recovery** and configure the following properties.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 17]
```diff
@@ -0,0 +1,534 @@
...
+    <table>
+        <thead>
+            <tr class="header">
+                <th>Field</th>
+                <th>Description</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr class="odd">
+                <td>Enable Notification Based Password Recovery</td>
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:157]:
```suggestion
                <td>Enable Notification-Based Password Recovery</td>
```

Comment (nilminiwso2) [reply en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:157]: depends on what's on the UI

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 18]
```diff
@@ -0,0 +1,534 @@
...
+                <td>Enable password recovery with notifications for the current tenant domain.</td>
+            </tr>
+            <tr class="odd">
+                <td>Enable Security Question Based Password Recovery</td>
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:161]:
```suggestion
                <td>Enable Security Question-Based Password Recovery</td>
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 19]
```diff
@@ -0,0 +1,534 @@
...
+                <td>Enable security question-based password recovery.</td>
+            </tr>
+            <tr class="odd">
+                <td>Enable Internal Notification Management</td>
+                <td>
+                    <p>
+                        Select to configure WSO2 Identity Server to send confirmation emails to the user.
+                        If the client application handles notification sending already, unselect it.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:169]:
```suggestion
                        If the client application handles notification sending already, clear this checkbox.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 20]
```diff
@@ -0,0 +1,534 @@
...
+                <td>Notify when Recovery Success</td>
+                <td>Send a notification when password reset is successful.</td>
+            </tr>
+            <tr class="odd">
+                <td>Recovery Link Expiry Time</td>
+                <td>
+                    <ul>
+                        <li>Validity period of the password reset email link.
+                        <li>Unit : <code>minutes</code>
+                        <li>Default : <code>1440 minutes = 1 day</code>
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:183]: If we put this in a code block, it is the actual value.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 21]
```diff
@@ -0,0 +1,534 @@
...
+                    </ul>
+                </td>
+            </tr>
+            <tr class="odd">
+                <td>SMS OTP Expiry Time</td>
+                <td>
+                    <ul>
+                        <li>Validity period of the password reset OTP.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:191]: same comment as before

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 22]
```diff
@@ -0,0 +1,534 @@
...
+    see how to [customize automated emails]({{base_path}}/guides/tenants/customize-automated-mails).
+
+### Manage SMS notification templates  
+  
+The templates for SMS notifications are stored in the registry. Follow the steps below to edit the existing
+sms notification templates.
+
+1. Log in to the Management Console and go to **Main > Registry> Browse**.
+
+2. On the **tree view** tab, click **system > config > identity > sms**. This will display all the available SMS notification templates.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:219]:
```suggestion
2. On the **tree view** tab, click **system > config > identity > sms**. This displays all the available SMS notification templates.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 23]
```diff
@@ -0,0 +1,534 @@
...
+
+!!! tip
+    The **passwordreset**, **passwordresetsucess**, and **resendpasswordreset** templates are used to send SMS notifications. You can edit and customize the SMS templates. <!--For more information, see [Managing SMS Notification Templates From the Registry](../../learn/managing-sms-templates-from-the-registry).-->
+
+## Try it out
+
+!!! info "Notification mechanisms"
+    WSO2 Identity Server provides the functionality to receive account recovery notifications internally or externally.
+
+    1. **Internal Notification Management**: Notification sending is managed by WSO2 Identity Server.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:237]:
```suggestion
    - **Internal Notification Management**: Notification sending is managed by WSO2 Identity Server.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 24]
```diff
@@ -0,0 +1,534 @@
...
+        
+    2. **External Notification Management**: Notification sending is managed by an external notification management mechanism.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:239]:
```suggestion
    - **External Notification Management**: Notification sending is managed by an external notification management mechanism.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 25]
```diff
@@ -0,0 +1,534 @@
...
+    
+    !!! note
+        To configure external notification management, `disable` the `Enable Internal Notification Management` property in `Account Recovery` configurations.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:242]:
```suggestion
        To configure external notification management, `disable` the `Enable Internal Notification Management` property in the `Account Recovery` configurations.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 26]
```diff
@@ -0,0 +1,534 @@
...
+        ![disable-internal-notifications]({{base_path}}/assets/img/guides/disable-internal-notification.png)
+
+### Create a user for recovery
+
+!!! tip "Before you begin"
+    Make sure you have a user with Email or Mobile configured. If you already have a user, skip to the next heading. If not, follow the steps below to create a new user and assign notification channels.
+
+1. Log in to the Management Console and click **Main -> Identity -> Users and Roles -> Add**.
+
+2. Click on **Add New User** and enter user credentials.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:253]:
```suggestion
2. Click **Add New User** and enter user credentials.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 27]
```diff
@@ -0,0 +1,534 @@
...
+
+    !!! Note
+        For more details on creating users and roles, see [Adding Users and Roles]({{base_path}}/references/concepts/user-management/user-management/)
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:256]:
```suggestion
        For more details on creating users and roles, see [Adding Users and Roles]({{base_path}}/references/concepts/user-management/user-management/).
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 28]
```diff
@@ -0,0 +1,534 @@
...
+
+3. Navigate to **Users and Roles > List > Users**.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:258]:
```suggestion
3. Go to **Users and Roles > List > Users**.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 29]
```diff
@@ -0,0 +1,534 @@
...
+
+4. Find the user from the list and click **View Roles**.
+
+5. Click on **Permissions** to edit the default permissions.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:262]:
```suggestion
5. Click **Permissions** to edit the default permissions.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 30]
```diff
@@ -0,0 +1,534 @@
...
+    ![add-login-permissions]({{base_path}}/assets/img/guides/user-permissions.png)
+
+6. From the list of permissions select **Login** permission and click **Update**.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:265]:
```suggestion
6. From the list of permissions, select **Login** permission and click **Update**.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 31]
```diff
@@ -0,0 +1,534 @@
...
+
+    !!! warning
+        This will update the permissions of the role. Therefore, all the users with the current role will receive login permissions.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:268]:
```suggestion
        This updates the permissions of the role. Therefore, all the users with the current role will receive login permissions.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 32]
```diff
@@ -0,0 +1,534 @@
...
+
+7. Go to **Users and Roles > List > Users**, click on **User Profile**, and update the Email and Mobile of the user.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:270]:
```suggestion
7. Go to **Users and Roles > List > Users**, click **User Profile**, and update the email and mobile of the user.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 33]
```diff
@@ -0,0 +1,534 @@
...
+          "rel": "next",
+          "href": "/t/carbon.superidentity/recovery/v0.9/security-question?username=sominda2",
+          "type": "GET"
+        }
+      ]
+    }
+    ]
+    ```
+
+    !!! note
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:322]: let's merge the two notes into one.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 34]
```diff
@@ -0,0 +1,534 @@
...
+         {
+            "rel": "resend",
+            "href": "/t/carbon.super/api/users/v1/recovery/password/resend",
+            "type": "POST"
+         }
+      ]
+    }
+    ```
+
+3. If you want to resend the notifications to the user via the notified channel in the above step, use the `resendCode` with the resend notifications API.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:355]:
```suggestion
3. If you want to resend notifications to the user via the notified channel in the above step, use the `resendCode` with the resend notifications API.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 35]
```diff
@@ -0,0 +1,534 @@
...
+        "code": "PWR-02005",
+        "message": "Successful password reset."
+    }
+    ```
+
+### With external notification management
+
+1. Use the following command to create a username recovery request.
+
+    ```java tab="Request"
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:418]: We must update these Request/Response code samples to the following format.

https://is.docs.wso2.com/en/6.0.0/guides/login/oidc-basic-client-profile/#try-out-the-flow

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 36]
```diff
@@ -0,0 +1,534 @@
...
+            "rel": "next",
+            "href": "/t/carbon.super/api/users/v1/recovery/username/recover",
+            "type": "POST"
+          }
+      ]
+    }]
+    ```
+
+    !!! note
+            The validity period of the recovery code is determined by the [2nd step of Configuring Password Recovery](#configuring-password-recovery)
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:446]:
```suggestion
            The validity period of the recovery code is determined by the [2nd step of configuring password recovery](#configuring-password-recovery)
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md (hunk 8 above)
Comment (divyaamunugama) [reply en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:81]: I think this is correct. 

shown below is how its represented in the docs:

image

Comment (divyaamunugama) [reply en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:81]: But +1 to your suggestion as well. Will update it

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 37]
```diff
@@ -0,0 +1,561 @@
...
+
+The password recovery flow of WSO2 Identity Server is as follows:
+
+1. User provides a set of unique claims to identify the user account.
+2. The API prompts the user to select a channel from the available notification channels for receiving the username recovery notification.
+3. The user selects a notification channel.
+4. The server sends the recovery notification to the user via the preferred notification channel.
+5. If the preferred channel is:
+    - **Email**: The user can click the reset link sent to the registered email address and reset the password.
+    - **SMS**: The user provides the received One-Time Password (OTP) and reset the password.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:18]:
```suggestion
    - **SMS**: The user provides the received One-Time Password (OTP) and resets the password.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 38]
```diff
@@ -0,0 +1,561 @@
...
+
+The following sections walk you through configuring and trying out password recovery via SMS and Email channels.
+
+??? Warning "If you are migrating from IS 5.10.0 or above"
+    If you have migrated from IS 5.10.0 or above, ensure that the `IdentityMgtEventListener` with the `orderId=50` is set to `false` and that the identity listeners with `orderId=95` and `orderId=97` are set to `true` in the `<IS_HOME>/repository/conf/deployment.toml` file.
+
+    Note: You can skip this step if there are no entries for `event.default_listener.xxx` in the `deployment.toml` file.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:25]:
```suggestion
    You can skip this step if there are no entries for the `event.default_listener.xxx` parameter in the `deployment.toml` file.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 39]
```diff
@@ -0,0 +1,561 @@
...
+    username= "wso2iamtest"
+    password= "Wso2@iam70"
+    hostname= "smtp.gmail.com"
+    port= 587
+    enable_start_tls= true
+    enable_authentication= true
+    ```
+
+    !!! Note
+        - Add this configuration only if you wish to configure WSO2 Identity Server to send confirmation emails. See [Configure the Email Sending Module]({{base_path}}/deploy/configure-email-sending/) for more information
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:57]:
```suggestion
        - Add this configuration only if you wish to configure WSO2 Identity Server to send confirmation emails. See [Configure the Email Sending Module]({{base_path}}/deploy/configure-email-sending/) for more information.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 40]
```diff
@@ -0,0 +1,561 @@
...
+            Validity period of SMS OTP when the selected channel is <code>SMS</code>.
+            <ul>
+                <li>Unit : <code>minutes</code>
+                <li>Default value : <code>1</code>
+           </ul>
+         </td>
+      </tr>
+    </table>
+
+3. Add an event publisher to `<IS_HOME/repository/deployment/server/eventpublishers`. You can use the following sample publisher to call a REST service to send confirmation codes. This sample uses the `http output adapter`.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:115]:
```suggestion
3. Add an event publisher to `<IS_HOME/repository/deployment/server/eventpublishers`. 

      You can use the following sample publisher to call a REST service to send confirmation codes. This sample uses the `http output adapter`.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 41]
```diff
@@ -0,0 +1,561 @@
...
+        For more details on creating users and roles, see [Adding Users and Roles]({{base_path}}/references/concepts/user-management/user-management/).
+
+3. Go to **Users and Roles > List > Users**.
+
+4. Find the user from the list and click **View Roles**.
+
+5. Click **Permissions** to edit the default permissions.
+    ![add-login-permissions]({{base_path}}/assets/img/guides/user-permissions.png)
+
+6. From the list of permissions, select **Login** permission and click **Update**.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:264]:
```suggestion
6. From the list of permissions, select the **Login** permission and click **Update**.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 42]
```diff
@@ -0,0 +1,561 @@
...
+            "type": "GET"
+            }
+        ]
+        }
+        ]
+        ```
+
+    !!! note
+        - The validity period of the recovery code is determined by the second step of [configuring password recovery](#set-up-password-recovery).  
+        - To try out password recovery with a challenge question, refer [Configuring Password Reset with Challenge Questions]({{base_path}}/guides/password-mgt/challenge-question/).
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:326]:
```suggestion
        - To try out password recovery with a challenge question, see [Configuring Password Reset with Challenge Questions]({{base_path}}/guides/password-mgt/challenge-question/).
```

**Review by nilminiwso2 — DISMISSED** [review]

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**